src/test_utils/fixtures.ts

```diff
@@ -0,0 +1,118 @@
+import { Server } from 'node:http';
+import { fork } from 'node:child_process';
+import { AddressInfo } from 'node:net';
+import { join } from 'node:path';
+import { expect, test as originalTest } from 'vitest';
+import strip from 'strip-ansi';
+import emocks from 'emocks';
+import express from 'express';
+
+import { log } from '../log';
+
+type Fixtures = {
+  runMockServer: (mockDir: string) => Promise<Server>;
+  testCliCommand: (
+    args: string[],
+    expected?: {
+      snapshot?: true;
+      stderr?: string | ReturnType<typeof expect.stringMatching>;
+      code?: number;
+    },
+  ) => Promise<void>;
+};
+
+export const test = originalTest.extend<Fixtures>({
+  // eslint-disable-next-line no-empty-pattern
+  runMockServer: async ({}, use) => {
+    let server: Server;
+    await use(async (mockDir) => {
+      const app = express();
+      app.use(express.json());
+      app.use(
+        '/',
+        emocks(join(process.cwd(), 'mocks', mockDir), {
+          '404': (_req, res) => res.status(404).send({ message: 'Not Found' }),
+        }),
+      );
+
+      await new Promise<void>((resolve) => {
+        server = app.listen(0, () => {
+          resolve();
+          log.debug(
+            'Mock server listening at %d',
+            (server.address() as AddressInfo).port,
+          );
+        });
+      });
+
+      return server;
+    });
+    await new Promise<void>((resolve, reject) =>
+      server.close((err) => (err ? reject(err) : resolve())),
+    );
+  },
+  testCliCommand: async ({ runMockServer, task }, use) => {
+    const mockDirOverride = task.name.match(/\*mockDir:(.*?)\*/);
+    const server = await runMockServer(
+      mockDirOverride ? mockDirOverride[1] : 'main',
+    );
+    await use(async (args, expected = { snapshot: true }) => {
+      let output = '';
+      let error = '';
+
+      const cp = fork(
+        join(process.cwd(), './dist/index.js'),
+        [
+          '--api-host',
+          `http://localhost:${(server.address() as AddressInfo).port}`,
+          '--output',
+          'yaml',
```

Comment:
Not related to your changes, but the default value is  `--output=table` which displays the command output in table output. (And we do some field-filtering for table output) 

Maybe worth having some tests with `--output table`?

Reply:
There's a `writer.test.ts` that does make snapshots of different output formats.

I believe we should stick to one format in other tests (preferrably the one that creates the minimal diff when changing data and easily human-readable - and I guess that's yaml).

Reply:
> I believe we should stick to one format in other tests (preferrably the one that creates the minimal diff when changing data and easily human-readable - and I guess that's yaml).

Totally agree with sticking to one format, as we have some tests for other formats.